ee_tests/src/spec_strategy/deployment_spec_strategy.ts

```diff
@@ -0,0 +1,131 @@
+import { SpacePipelinePage } from '../page_objects/space_pipeline.page';
+
+export abstract class DeploymentStrategy {
+
+    protected spaceName: string;
+
+    protected spacePipelinePage: SpacePipelinePage;
+
+    protected spaceDeploymentsPage: SpaceDeploymentsPage;
+
+    constructor(spaceName: string, spacePipelinePage: SpacePipelinePage) {
+        this.spaceName = spaceName;
+        this.spacePipelinePage = spacePipelinePage;
+        this.spaceDeploymentsPage = new SpaceDeploymentsPage();
+    }
+
+    public async showDeploymentsScreen() {
+        support.info('Verifying deployments page');
+        await this.spacePipelinePage.spaceHeader.deploymentsOption.clickWhenReady();
+        await browser.sleep(5000);
+        this.spaceDeploymentsPage = new SpaceDeploymentsPage();
+    }
+
+    public async verifyApplication(): Promise<DeployedApplication> {
+        support.info('Verifying deployed applications');
+        let applications = await this.spaceDeploymentsPage.getDeployedApplications();
+        expect(applications.length).toBe(1, 'number of deployed applications');
+
+        let application = applications[0];
+        expect(application.getName()).toBe(this.spaceName, 'application name');
+        return application;
+    }
+
+    public async verifyEnvironments(application: DeployedApplication) {
+        support.info('Verifying application\'s environments');
+        let environments = await application.getEnvironments();
+        expect(environments.length).toBe(2, 'number of environments');
+        await this.testEnvironmentsInternal(environments);
+    }
+
+    public async verifyResourceUsage() {
+        support.info('Verifying resources usage');
+        let data = await this.spaceDeploymentsPage.getResourceUsageData();
+        expect(data.length).toBe(2, 'there should be stage and prod environment');
+        await this.verifyResourceUsageInternal(data);
+    }
+
+    protected abstract async testEnvironmentsInternal(environments: DeployedApplicationEnvironment[]): Promise<void>;
+
+    protected abstract async verifyResourceUsageInternal(data: ResourceUsageData[]): Promise<void>;
+}
+
+export class ReleaseDeploymentStrategy extends DeploymentStrategy {
+
+    protected async testEnvironmentsInternal(environments: DeployedApplicationEnvironment[]) {
+        // nothing to test here
+    }
+
+
+    protected async verifyResourceUsageInternal(data: ResourceUsageData[]) {
+        // nothing to test here
+    }
+}
+
+export class StageDeploymentStrategy extends ReleaseDeploymentStrategy {
+    protected async testEnvironmentsInternal(environments: DeployedApplicationEnvironment[]) {
+        await super.testEnvironmentsInternal(environments);
+        support.info('Verifying application\'s stage environment');
+
+        let environment = environments[Environment.STAGE];
```

Comment:
Is the `Environment` the same thing as a `ReleaseStrategy`?

Reply:
Those are 2 independent things.